### PR TITLE
[issue 105] - Adding the RHSSO Template provisioner

### LINKED ISSUE #105: Add RHSSO 7.6.z Template based provisioning tooling
**Is your feature request related to a problem? Please describe.**
A template based provisioner for RHSSO 7.4.z is needed to run interoperability tests that involve RHSSO 7.4.z.  
Some 

**Describe the solution you'd like**
- [ ] Add the provisioning tooling (factory, application interface, provisioner), as done for the EAP 7.z templates, see #80 
- [ ] Add provisioner tests (lifecycle, cleanup)
- [ ] Add properties to the e2e-prod CI script

**Describe alternatives you've considered**
Rework _internal_ tests that are currently using RHSSO Template based provisioner in order to use a different startegy but that would require for more work and time.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
Fix #105 - ported to support testing of RH-SSO via templates, which is still being used by internal pipelines.

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	 * Realm configuration in json format for Keycloak partial import
+	 * <p>
+	 * https://www.keycloak.org/docs/9.0/server_admin/index.html#importing-a-realm-from-exported-json-file
+	 * <p>
+	 * Requires template parameters SSO_REALM, SSO_SERVICE_USERNAME, SSO_SERVICE_PASSWORD to be set
+	 *
+	 * @return Instance of {@link Path} representing a YAML definition for the desired realm configuration
+	 */
+	default Path getRealmConfigurationFilePath() {
+		return null;
```
Comment (tommaso-borgato) [inline tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java:41]: @fabiobrz importing a realm isn't needed, right?

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java [hunk 2]
```diff
@@ -62,6 +62,7 @@ public KeycloakAdminClient(final String url, final String realm, final String us
 	 */
 	public void importRealmConfiguration(InputStream is) throws IOException {
 		PartialImportRepresentation piRep = JsonSerialization.readValue(is, PartialImportRepresentation.class);
+		// TODO - fix, the realm name should be passed in as a parameter
 		keycloak.realm("wildfly-realm").partialImport(piRep);
```
Comment (tommaso-borgato) [inline tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java:66]: @fabiobrz isn't the realm name specified in the JSON file to import?

**Comment by tommaso-borgato** [comment]
@fabiobrz LGTM, I just left  couple of questions...

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java (hunk 2 above)
Comment (fabiobrz) [reply tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java:66]: Now that you've mentioned it, I see I have been maybe a bit too quick in here.
There's actually some controversial issue around the Keycloak/RH-SSO admin client (also about the compatible versions), and I am going to check it, thanks for bringing it up.

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java (hunk 1 above)
Comment (fabiobrz) [reply tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java:41]: It depends, there are internal tests relying on this specific provisioner, hence I am going to check and get beck here, thanks for now!

Comment (fabiobrz) [reply tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java:41]: Confirmed, it is used by existing EAP7 + RHSSO tests.

File: tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java (hunk 2 above)
Comment (fabiobrz) [reply tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java]: Back with some updates. So the answer is "no", meaning that the Keycloak Admin API contracts require for the realm name, to identify the realm where resources will be created.
Here's an example from an existing test:

```
{
  "roles": {
    "realm" : [{
      "name": "user"
    }]
  },

  "users": [{
    "username" : "user",
    "enabled": true,
    "credentials" : [{
      "type" : "password",
      "value" : "password"
    }],
    "realmRoles": ["user"],
    "clientRoles": {
      "account": ["view-profile", "manage-account"]
    }
  }]
}
```
Anyway, I fixed the method as well, according to the `TODO`. Thanks again for bringing this up.

**Comment by fabiobrz** [comment]
> @fabiobrz LGTM, I just left couple of questions...

Thanks @tommaso-borgato - I addressed your concerns, feel free to let me know in case you think something else is needed here.

**Pull request merged**